### PR TITLE
Fix delete record may be re-applied

### PR DESCRIPTION
**Pull request by yah01**
/kind bug
related #24234 
- Also fixes the load stats never stops
- Also fixes GetDeletedCount() returns the number of rows

### CODE REVIEW

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

File: internal/querynode/load_segment_task.go [hunk 1]
```diff
@@ -120,6 +120,26 @@ func (l *loadSegmentsTask) Execute(ctx context.Context) error {
...
+			if err != nil {
+				break
+			}
+			err = segment.FlushDelete()
+			if err != nil {
+				break
+			}
+		}
+
+		if err != nil {
```
Comment (congqixia) [inline internal/querynode/load_segment_task.go:135]: This error will never be not nil since err in for loop is shaded

**Review by congqixia — CHANGES_REQUESTED** [review]

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

File: internal/querynode/segment.go [hunk 2]
```diff
@@ -784,16 +791,26 @@ func (s *Segment) segmentDelete(entityIDs []primaryKey, timestamps []Timestamp)
...
 	s.mut.RLock()
 	defer s.mut.RUnlock()
 	if !s.healthy() {
 		return fmt.Errorf("%w(segmentID=%d)", ErrSegmentUnhealthy, s.segmentID)
 	}
 
-	return s.deleteImpl(entityIDs, timestamps)
-}
+	start := sort.Search(len(timestamps), func(i int) bool {
+		return timestamps[i] >= s.flushedDeletedTimestamp+1
```
Comment (congqixia) [inline internal/querynode/segment.go:805]: why not use 
```
return timestamps[i] > s.flushedDeletedTimestamp
```

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**